### PR TITLE
Homework1

### PR DESCRIPTION
**Pull request by mishakras**

### CODE REVIEW

File: homework1/task5.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+The written function should return the sum of this sub-array.
+Examples:
+    nums = [1, 3, -1, -3, 5, 3, 6, 7], k = 3
+    result = 16
+"""
+import math
+from typing import List
+
+
+def find_maximal_subarray_sum(nums: List[int], k: int) -> int:
```
[the remark this answers is not in the document]
Comment (mishakras) [reply homework1/task5.py:15]: Fixed, I assumed that order of elements of array doesn't matter.

File: homework1/task4.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+Given four lists A, B, C, D of integer values,
+    compute how many tuples (i, j, k, l) there are
+    such that A[i] + B[j] + C[k] + D[l] is zero.
+We guarantee, that all A, B, C, D have same length of
+    N where 0 ≤ N ≤ 1000.
+"""
+from typing import List
+
+
+def check_sum_of_four(a: List[int], b: List[int], c: List[int],
```
[the remark this answers is not in the document]
Comment (mishakras) [reply homework1/task4.py:12]: Added some check for early ending, best case is O(n*log (N)), worst still O(N^4).

File: homework1/task3.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
+"""
+Write down the function, which reads input line-by-line, and
+find maximum and minimum values.
+Function should return a tuple with the max and min values.
+For example for [1, 2, 3, 4, 5], function should return [1, 5]
+We guarantee, that file exists and contains line-delimited integers.
```
[the remark this answers is not in the document]
Comment (mishakras) [reply homework1/task3.py:6]: Fixed

File: homework1/task2.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+Given a cell with "it's a fib sequence" from slideshow,
+    please write function "check_fib", which
+    accepts a Sequence of integers, and
+    returns if the given sequence is a Fibonacci sequence
+We guarantee, that the given sequence contain >= 0 integers inside.
+"""
+from typing import Sequence
+
+
+def check_fibonacci(data: Sequence[int]) -> bool:
```
[the remark this answers is not in the document]
Comment (mishakras) [reply homework1/task2.py:11]: Do you want strict fibonacchi sequence, a.i. 1,1,2,3,5... or any that follows Yn=Y(n-1)+Y(n-2)? Later works.

[the remark this answers is not in the document]
Comment (nikitosboroda) [reply homework1/task2.py:11]: strict
better 0, 1, 1, 2 ...

**Review by nikitosboroda — CHANGES_REQUESTED** [review]

**Review by nikitosboroda — APPROVED** [review]

**Pull request merged**